lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -283,9 +283,9 @@ def self.settings_for_dependent_targets(target, dependent_targets)
         #
         def self.add_language_specific_settings(target, xcconfig)
           if target.uses_swift?
-            build_settings = {
-              'OTHER_SWIFT_FLAGS' => '$(inherited) ' + quote(%w(-D COCOAPODS)),
-            }
+            other_swift_flags = ['$(inherited)', quote(%w(-D COCOAPODS))]
+            other_swift_flags += [quote(%w(-suppress-warnings))] if target.try(:inhibit_warnings?)
```

Comment:
can be `other_swift_flags << quote(%w(-suppress-warnings)) if target.inhibit_warnings?`